rainbow-me/rainbow
# Price widgets

This PR adds iOS widgets that display the current price and % price change (via CoinGecko) of a token selected by the user. The user can also (optionally) specify a currency that the price should be displayed in (USD, GBP, ETH etc.). If the user doesn't specify a display currency, it should default to the user's display currency settings in the Rainbow app.

All of these configurations can be modified by pressing on the widget for a few seconds and clicking on the button that says "Edit Widget"

QA Instructions:
- Ensure that the widget displays price, price change, token name, and time last updated at all times. If any of these are missing, it should fallback to a different screen that I created.
- All "Top Tokens" should have icons displayed
- Be on the lookout for any weird color pairings. These are meant to match the color scheme of the token icon but they're autogenerated so they could be off.
- The data displayed on the widget should refresh approx. every 10 min, and this should be reflected by the timestamp at the bottom of the widget.
- Switching the display currency should work for all the provided options, and should fallback to the user's corresponding setting in the Rainbow app if none are selected
- If the user changes their display currency setting inside the Rainbow app (not widget), the widget should reflect these changes the next time it updates (approx. every 10 min), unless a different display currency is specified for the widget.
